src/main/java/seedu/exercise/logic/commands/events/EventFactory.java

```diff
@@ -46,21 +51,54 @@ public static Event commandToEvent(UndoableCommand command) throws CommandExcept
         case ClearCommand.COMMAND_WORD:
             return new ClearEvent(((ClearCommand) command).getPayload());
 
+        case ScheduleCommand.COMMAND_WORD:
+            return generateEventFromScheduleCommand((ScheduleCommand) command);
+
+        case ResolveCommand.COMMAND_WORD:
+            return new ResolveEvent(((ResolveCommand) command).getPayload());
+
         default:
             throw new CommandException(
                     String.format(MESSAGE_COMMAND_NOT_UNDOABLE, commandWord));
         }
     }
 
+    /**
+     * Generates a schedule regime or schedule complete event based on the command type.
+     *
+     * @param command a {@link ScheduleCommand} to be represented with using an Event object
+     * @return a {@link ScheduleRegimeEvent} or a {@link ScheduleCompleteEvent}
+     * that can be undone or redone
+     */
+    private static Event generateEventFromScheduleCommand(ScheduleCommand command) throws CommandException {
+        String resourceType = command.getCommandTypeIdentifier();
+        EventPayload<Schedule> eventPayload;
+        switch (resourceType) {
+        case ScheduleRegimeCommand.UNIQUE_IDENTIFIER:
+            eventPayload = ((ScheduleRegimeCommand) command).getPayload();
+            return new ScheduleRegimeEvent(eventPayload);
+
+        case ScheduleCompleteCommand.UNIQUE_IDENTIFIER:
+            eventPayload = ((ScheduleCompleteCommand) command).getPayload();
+            return new ScheduleCompleteEvent(eventPayload);
+
+        default:
+            throw new CommandException(
+                    String.format(MESSAGE_UNIQUE_IDENTIFIER_NOT_FOUND,
+                            resourceType,
+                            command.getUndoableCommandWord()));
+        }
+    }
+
     /**
      * Generates a add exercise or add regime event based on the command type.
      *
-     * @param command a {@code AddCommand} to be represented with using an Event object
-     * @return an {@code AddExerciseEvent}, {@code AddRegimeEvent} or {@code EditRegimeEvent}
+     * @param command a {@link AddCommand} to be represented with using an Event object
+     * @return an {@link AddExerciseEvent}, {@link AddRegimeEvent} or {@link EditRegimeEvent}
      * that can be undone or redone
      */
     protected static Event generateEventFromAddCommand(AddCommand command) throws CommandException {
```

Comment:
The generate methods are all protected but the generateEventFromScheduleCommand is private?

Reply:
Changed already. Needed it to be package-private for testing